app/telemetry.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  startBackgroundUpdate,
+  trackCli,
+  setDefaultComponentId,
+  setGatsbyCliVersion,
+  trackFeatureIsUsed,
+} from "gatsby-telemetry"
+import { app, ipcMain, IpcMainEvent } from "electron"
+import { ITelemetryTagsPayload } from "gatsby-telemetry/lib/telemetry"
+import Store from "electron-store"
+
+interface IConfigType {
+  telemetryOptIn: boolean
+}
+
+let telemetryEnabled: boolean | undefined
+
+export function telemetryTrackFeatureIsUsed(name: string): void {
+  console.log(`track feature is used`, name, { telemetryEnabled })
+  if (telemetryEnabled) {
+    trackFeatureIsUsed(name)
+  }
+}
+
+export function trackEvent(
+  input: string | Array<string>,
+  tags?: ITelemetryTagsPayload
+): void {
+  console.log(`track event`, input, tags, { telemetryEnabled })
+  if (telemetryEnabled) {
+    trackCli(input, tags)
+  }
+}
+
+export function initializeTelemetry(): void {
+  setDefaultComponentId(`gatsby-desktop`)
+  setGatsbyCliVersion(app.getVersion())
+  startBackgroundUpdate()
+  const store = new Store<IConfigType>()
+  store.onDidChange(`telemetryOptIn`, (newValue?: boolean) => {
+    telemetryEnabled = !!newValue
+  })
+  telemetryEnabled = store.get(`telemetryOptIn`)
```

Comment:
wouldn't making `const telemetryEnabled = store.get("telemetryOptIn")` at line 39-40 help getting rid of the `let telemetryEnabled: boolean | undefined` variable? 

I mean, in other functions, we can also use `store.get("telemetryOptIn")` instead of the `telemetryEnabled` variable no?

Also, from the signature of `store.onDidChange`, it seems that this function runs _after_ the change has been made in the store so maybe we can even get rid of this listener if we can data from the store directly?